R/as-tibble.R

```diff
@@ -1,26 +1,56 @@
+#' Flatten a Crunch Cube
+#'
+#' Crunch Cubes can be expressed as a long data frame instead of a
+#' multidimensional array. In this form each dimension of the cube is a variable
+#' of the dataframe and the cube values is expressed columns for each measure.
+#' This is useful both to better understand what each entry of a cube
+#' represents, and to work with the cube result using tidyverse tools.
+#'
+#' @param x a CrunchCube
+#' @param return_real Do you want to return the tibble represenatation of the
+#'   real cube with MR selection dimensions and missing values. Mostly useful
+#'   for debugging complex cubes.
+#' @param ... futher arguments passed on to `dplyr::as_tibble()`
+#'
 #' @export
 #' @importFrom tibble as_tibble
-as_tibble.CrunchCube <- function (x, ...) {
+as_tibble.CrunchCube <- function (x, return_real = FALSE, ...) {
     ## TODO: Consider using `dplyr::tbl_cube` class
     ## TODO: NA behavior? Add way to get missings not missing. (This uses
     ## as.array to remove extraneous dims, and doing so applies the '@useNA'
```

Comment:
This is probably scope creep, but after looking through the path for real vs. user cube: the real cube always preserves all missing categories, where as the user one follows what's in `useNA`. I'm not certain that should hold up these improvements (especially since we don't have any evidence that anyone is using this already, nevermind would be confused by the fact that you get different missing behaviors!)